Core/src/main/java/com/paypal/android/core/graphql/common/GraphQlClient.kt

```diff
@@ -0,0 +1,46 @@
+package com.paypal.android.core.graphql.common
+
+import android.util.Log
+import com.paypal.android.core.APIClientError
+import com.paypal.android.core.GraphQlRequestFactory
+import com.paypal.android.core.Http
+import org.json.JSONObject
+import java.net.HttpURLConnection
+
+interface GraphQlClient {
+    suspend fun <T> executeQuery(query: Query<T>): GraphQlQueryResponse<T>
+}
+
+class GraphQlClientImpl : GraphQlClient {
+    private val http = Http()
+    private val graphQlRequestFactory = GraphQlRequestFactory()
+
+    override suspend fun <T> executeQuery(query: Query<T>): GraphQlQueryResponse<T> {
+        try {
+            val httpRequest = graphQlRequestFactory.createHttpRequestFromQuery(
+                query.requestBody()
+            )
+            val httpResponse = http.send(httpRequest)
+            val bodyResponse = httpResponse.body
```

Comment:
Do we get a body if there's a server error? Maybe we should do the status code check first and then this portion